backend/init_data/skills/dingtalk-knowledge/provider.py

```
@@ -0,0 +1,650 @@
+            )
+
+            if upload_result.exit_code != 0:
+                raise RuntimeError(
+                    f"Upload failed: {upload_result.stderr or 'Unknown error'}"
+                )
+
+            # Parse response
+            try:
+                api_response = json.loads(upload_result.stdout)
+            except json.JSONDecodeError as e:
+                raise RuntimeError(f"Failed to parse upload response: {e}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)